src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionDiagnosticAnalyzer.Analyzer.cs

```diff
@@ -122,7 +154,21 @@ private static AnalysisResult Aggregate<T>(SyntaxList<T> nodes, Func<AnalysisRes
 
             public override AnalysisResult VisitAssignmentExpression(AssignmentExpressionSyntax node)
             {
-                return AnalysisResult.Assignment(node.Kind(), node.Left);
+                if (DependsOnPreviousAssignments(node))
+                {
+                    return AnalysisResult.Failure;
+                }
+
+                var targets = _firstSwitchSection ? _originalTargets : _currentTargets;
+                targets.Add(node.Left);
+                return AnalysisResult.Assignment(node.Kind());
+            }
+
+            private bool DependsOnPreviousAssignments(AssignmentExpressionSyntax assignment)
+            {
+                return _originalTargets.Any(
+                    (target, right) => right.DescendantNodesAndSelf().Any(n => SyntaxFactory.AreEquivalent(target, n)),
```

Comment:
1. Is there any more efficient way to do this?
2. This is still not complete. In case of an array (for instance) use of the whole array is now unsafe, not that exact index. How we could check for such dependency?

Reply:
Note: if anyone feel like this wouldn't worth it, we can just not offer the fix for multi-assignments.

Reply:
> Note: if anyone feel like this wouldn't worth it, we can just not offer the fix for multi-assignments (It could greatly simplify the analyzer).

I believe the analyzer/fix for the single assignment case is most likely to be used and extremely valuable. I would prefer we do this simplification for the initial implementation, and add the multi-assignment complexity only if there are customer requests and we can be sure we are not silently breaking semantics with the code fix. I am also concerned about the readability of generated code for this form, not everyone would prefer assigning to a tuple with switch expression over a switch statement with multiple explicit assignments in each switch section.